experimental/javascript.html

```diff
@@ -0,0 +1,161 @@
+                '<label for="dim-col'+idd+'">Column Name: </label>'+
+                '<input type="text" name="dim-col" id="dim-col'+idd+'" class="width100" placeholder="column">'+
+              '</div>'+
+              '<button name="delete" class="red btn-delete1" onclick="removeElement(\''+dimensionId+'\')">DELETE</button>';
+    addElement('dimension','div',dimensionId,html);   //parentId, elementTag, elementId, html
+  }
+    
+  function funcAddDate(button){
+    var text=button.textContent;
+    if(text=="ADD"){
+      var html= '<div class="block">'+
+                  '<label for="date-col">Date Column:&ensp;</label>'+
+                  '<input type="text" id="date-col" name="date-col" class="width130" placeholder="column"> '+
+                '</div>'+
+                '<div class="block">'+
+                  '<div class="inline form-group">'+
+                    '<label for="date-start">Start Date:</label>'+
+                    '<input type="date" id="date-start" name="date-start" class="width125">'+
+                  '</div>'+
+                  '<div class="inline form-group">'+
+                    '<label for="date-end">End Date:</label>'+
+                    '<input type="date" id="date-end" name="date-end" class="width125">'+
+                  '</div>'+
+                '</div>';
+      addElement('date','div',"date-range",html);   //parentId, elementTag, elementId, html
+      button.classList.remove("share","btn-add");
+      button.classList.add("red","btn-delete2");
+      button.textContent="DELETE";
+    }
+    else{
+      removeElement("date-range");
+      button.classList.remove("red","btn-delete2");
+      button.classList.add("share","btn-add");
+      button.textContent="ADD";
+    }
+  }
+    
+  function funcAddGranularity(button){
+    var text=button.textContent;
+    if(text=="ADD"){
+      var html= '<div class="drp1">'+
+                  '<div class="drp2">'+
+                    '<select id="period">'+
+                      '<option>Hourly</option>'+
+                      '<option>Daily</option>'+
+                      '<option>Fortnightly</option>'+
+                      '<option>Monthly</option>'+
+                      '<option>Quarterly</option>'+
+                      '<option>Semi-Annually</option>'+
+                      '<option>Annually</option>'+
+                    '</select>'+
+                  '</div>'+
+                '</div>';
+      addElement('granularity','div',"grn",html);   //parentId, elementTag, elementId, html
+      button.classList.remove("share","btn-add");
+      button.classList.add("red","btn-delete2");
+      button.textContent="DELETE";
+    }
+    else{
+      console.log("removing");
+      removeElement("grn");
+      button.classList.remove("red","btn-delete2");
+      button.classList.add("share","btn-add");
+      button.textContent="ADD";
+    }
+  }
+    
+  function addElement(parentId, elementTag, elementId, html) {
```

Comment:
Can move these kind of helper functions to separate file called helper.gs.

Reply:
Will do that